From 8ed0402c8d4388ba82e3fa879a73f6096e1da6b3 Mon Sep 17 00:00:00 2001
From: "Diwakar, Praveen" <praveen.diwakar@intel.com>
Date: Wed, 28 Jun 2017 16:49:23 +0530
Subject: [PATCH 0756/2367] ASoC: Intel: Skylake: Get dsp core count from hw
 config ipc

Instead of getting dsp core count from hard coded value, use hw
config ipc to reterive the same.

Signed-off-by: Diwakar, Praveen <praveen.diwakar@intel.com>
---
 sound/soc/intel/skylake/cnl-sst.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/sound/soc/intel/skylake/cnl-sst.c b/sound/soc/intel/skylake/cnl-sst.c
index 67c3da0..05b8a7b 100644
--- a/sound/soc/intel/skylake/cnl-sst.c
+++ b/sound/soc/intel/skylake/cnl-sst.c
@@ -272,6 +272,16 @@ static int cnl_load_base_firmware(struct sst_dsp *ctx)
 		goto cnl_load_base_firmware_failed;
 	}
 
+	ret = skl_get_hardware_configuration(ctx);
+	if (ret < 0) {
+		dev_err(ctx->dev, "hwconfig ipc failed !\n");
+		ret = -EIO;
+		goto cnl_load_base_firmware_failed;
+	}
+
+	/* Update dsp core count retrieved from hw config IPC */
+	cnl->cores.count = cnl->hw_property.dsp_cores;
+
 	return 0;
 
 cnl_load_base_firmware_failed:
-- 
2.7.4

